### DISCUSSION TITLE
Proton Pass export does not import username / email field to Vaultwarden

### DISCUSSION THREAD

**Discussion by br4yd** [dani-garcia/vaultwarden#5165] [Q&A] [answered]
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.32.3
* Web-vault version: v2024.6.2c
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Debian)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.46.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 

```json
{
  "_duo_akey": null,
  "_enable_duo": true,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_max_note_size": 10000,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://******************",
  "domain_origin": "*****://******************",
  "domain_path": "",
  "domain_set": true,
  "duo_context_purge_schedule": "30 * * * * *",
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "duo_use_iframe": false,
  "email_2fa_auto_fallback": false,
  "email_2fa_enforce_on_verified_invite": false,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "enable_websocket": true,
  "enforce_single_org_with_reset_pw_policy": false,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "http_request_block_non_global_ips": true,
  "http_request_block_regex": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "increase_note_size_limit": false,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 600000,
  "push_enabled": true,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "******************",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "********************",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "************************",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

### Vaultwarden Build Version

1.32.3

### Deployment method

Official Container Image

### Reverse Proxy

nginx

### Host/Server Operating System

Linux

### Clients

Web Vault, Browser Extension, Desktop

### Steps To Reproduce

1. Export JSON / CSV from Proton Pass
2. You will receive a json or csv that has fields for username and email (Proton Pass can have both in the same entry)
3. Import into Vaultwarden
4. Everything in the import is fine besides the username field. This will be empty every time. Usually you'd want to take the email field from Proton Pass for the username if either email exists or email and username exists. And the username field in case the email field is empty.

### Expected Result

Email and Username field filled -> Email should be imported
Email filled and Username empty -> Email should be imported
Email empty and Username filled -> Username should be imported
Both empty -> Empty field

### Actual Result

Neither the username nor the email is being imported causing useless imports.

**Comment by BlackDex** [comment]
Vaultwarden doesn't handle the import it self. Imports are handled by the clients which sends them encrypted to the server.

I would suggest to try this on the Bitwarden Cloud environment and if it breaks there too, report it at Bitwarden. If it doesn't, then it's probably fixed in a newer web-vault which we do not yet support.

An other option is to see if you can export it in an other format, or convert it your self to something else.

**Comment by tumatanquang** [comment]
@BlackDex This is probably a Vaultwarden issue. I tried importing on Bitwarden Web vault and the import is fine.
However, its behavior is not quite right, specifically if:
Username and email data both exist: Using email data as username field value, username data is ignored.
If only one of username or email data exists: It will be imported into username field (works correctly).
If neither data exists: It will be ignored.

**Reply by BlackDex** [reply]
As mentioned in my previous post. If it works on Bitwarden Cloud, then it is probably fixed in a web-vault not yet supported by Vaultwarden. Since this isn't a specific issue tied to Vaultwarden it self, in the sense it is a bug in the Vaultwarden code, i do not deem this as a Vaultwarden specific issue.

I am working on getting a newer web-vault working without to many difficulties and backwards compatibility, but that takes a bit time unfortunately.

Also, if you say the Bitwarden Cloud also doesn't do it fully correctly, that isn't something we can change of course. And when Vaultwarden will have a newer web-vault, that specific quirk will then also be for Vaultwarden. So, i would suggest to report that at least at Bitwarden and do not mention Vaultwarden unless you want the report to be dismissed directly.

**Reply by tumatanquang** [reply]
I posted a thread asking Bitwarden to handle the data more accurately and made no mention of Vaultwarden.

**Reply by tumatanquang** [reply]
@BlackDex According to this answer: https://community.bitwarden.com/t/do-not-skip-the-username-data-field-when-importing-data-from-proton-pass/75498/2

The JSON import function from ProtonPass has not changed since August 9, 2024 and I also tried importing the data again on Bitwarden Web vault, the data was put into the correct fields.
So this issue has been fixed on Bitwarden for a long time but not on Vaultwarden.

**Reply by BlackDex** [reply, marked as answer]
Yes as mentioned before, Vaultwarden is still on an older version of the web-vault because of some changes Bitwarden made, which Vaultwarden doesn't get support.

I'm working on this but that takes time. Once we are able to use that newer web-vault it will work again.

**Reply by nitanmarcel** [reply]
Any update on this or workarounds?

**Reply by BlackDex** [reply]
try the `testing` tagged images of Vaultwarden, they have web-vault v2025.1.0 right now.
There are some org quirks though, but other than that it seems to work fine

**Reply by nitanmarcel** [reply]
> try the `testing` tagged images of Vaultwarden, they have web-vault v2025.1.0 right now. There are some org quirks though, but other than that it seems to work fine

ah well, right after I posted this message it came in my mid to import them in the official vault, export them and import them in vaultwarden which worked xD

**Reply by xenoplayed** [reply]
I ran into the same problem with importing from proton and just noticed, that the current Bitwarden Desktop App (macOS, Version 2025.1.0 (36032)) already imports all values correct 🥳
